# Twitter Sentiment Analysis - Alisha Siddiqui

###  HW 1  - Using tweets with the word 'trump' in it. 


#### 1. Data collection was done using a developer twitter account and running tweetering.py in spyder####

In [95]:
import re #Performing data manipulation

Executing Data cleaning using lower, strip and replace functions

In [96]:
data_clean=[] #creating an empty list
file=open("Trump.txt","r") #open the Trump.txt file in read mode to analyze the twitter data.
##Looping through the file to convert the strings to lowercase,remove all leading and trailing white spaces.
for w in file: 
    w=w.lower()
    w=w.strip()
    w=w.replace("@","").replace(":","").replace('rt',"").replace(".","").\
    replace(",","").replace("!","").replace("?","").replace("#","").replace("\\","").\
    replace("\n","").replace("%","").replace("_","").replace("|","").replace("n\'","").\
    replace("-","").replace(')',"").replace("(","")
   
    for w in w.split():
       data_clean.append(w) #after splitting the strings append it to the data_clean list. 


In [97]:
#For further cleaning the data, using re.search function  
#creating a temporary empty list
a=[] 
#looping through the data which is cleaned above
for i in range(len(data_clean)): 
#re.search function will search for the mentioned characters
    if(re.search(r"https|xe2|xf0|[0-9]",data_clean[i]) is not None):    
#if the match is true(not None)]will append the location of item to list a
      a.append(i)

In [98]:
#Using an if condition to not include the items with location in list a while looping through data
data_clean=[data_clean[i] for i in range(len(data_clean)) if (i not in a)] 

In [99]:
print(data_clean[:100])

["b'", 'nanniquilts', 'brhodes', 'lawrence', 'ah', 'but', 'the', 'grifter', 'family', 'is', 'having', 'a', 'great', 'time', 'on', 'our', 'dime', 'how', 'is', 'this', 'for', 'impeachment', 'day', 'subduedradical', 'fakenewscnn', 'lowest', 'viewersnfakenewsmsnbc', 'lowest', 'viewers', 'nbiasmedia', 'brithume', 'marcthiessen', 'because', 'when', 'the', 'impeachment', 'was', 'in', 'the', 'house', 'bolton', 'originally', 'said', 'he', 'wouldt', 'democrats', 'say', 'there', 'is', '"overwhelming"', 'evidence', 'against', 'trumpnnyet', 'after', 'weeks', 'of', 'an', 'impeachment', 'inquirynnthousands', 'of', 'santiagomayer', 'a', 'picture', 'of', 'lev', 'parnas', 'a', 'possible', 'witness', 'with', 'pam', 'bondi', 'one', 'of', 'be', 'a', 'damn', 'shame', 'jaketapper', 'mffisher', 'zeitchikwapo', 'saudi', 'phone', 'because', 'trump', 'like', "himb'", 'tribelaw', 'turns', 'out', 'we', 'have', 'a', 'really', 'deep', 'bench', 'guys', 'every', 'one', 'of']


#### 2. Using text files(with negative, positive and stop words) for comparison with cleaned data. 

In [100]:
positive =[]
f1= open('positive.txt','r')
for w in f1:
    w=w.lower()
    w=w.strip()
    positive.append(w)
len(positive)
positive
f1.close()

In [101]:
negative = []
f2=open('negative1.txt','r')
for x in f2:
    x=x.lower()
    x=x.strip()
    negative.append(x)
len(negative)
negative
f2.close()

In [102]:
stop=[]
f3 = open('stopwords1.txt','r')
for y in f3:
    y = y.lower()
    y = y.strip()
    stop.append(y)
len(stop)
stop
f3.close()

#### 3. Comparing the lists above and performing analysis with the cleaned data

In [103]:
#Initializing counter with 0 that will be used to count the no of +ve, -ve, others and stop words.
stop_counter = 0
negative_counter = 0
pos_counter = 0
positive_counter = 0
others_counter = 0
total_count = 0

In [104]:
#Looping through the files to count the match and increase the counter by 1 respectively 
for w in data_clean:
    if w in positive:
        pos_counter +=1 
    elif w in negative:
            negative_counter +=1
    elif w in stop:
            stop_counter+=1
    else:
        others_counter+=1

In [105]:
for w in data_clean:
    if w in positive and w!="trump":
        positive_counter +=1

In [106]:
total_count = positive_counter+negative_counter+stop_counter+others_counter
print("Total word count =%.2f" %(total_count))
print("positive word count =%d" %positive_counter)
print("negative word count =%d" %negative_counter)
print("stop word count =%d" %stop_counter)
print("other word count = %d" %others_counter)


print("Ratio of positive to  total word count = %.2f%%" %((positive_counter/total_count)*100))
print("Ratio of negative to  total word count = %.2f%%" %((negative_counter/total_count)*100))
print("Ratio of stopwords to total word count = %.2f%%" %((stop_counter/total_count)*100))

print("Ratio of other word v/s total words ratio = %.2f" %(others_counter/(total_count)))

print("Sum of all positive/negative words = %.2f" %(positive_counter+negative_counter))

print("Ratio of positive/(postive + negative words) = %.2f" %(positive_counter/(positive_counter+negative_counter)))



Total word count =55075.00
positive word count =1279
negative word count =2126
stop word count =22115
other word count = 29555
Ratio of positive to  total word count = 2.32%
Ratio of negative to  total word count = 3.86%
Ratio of stopwords to total word count = 40.15%
Ratio of other word v/s total words ratio = 0.54
Sum of all positive/negative words = 3405.00
Ratio of positive/(postive + negative words) = 0.38


### We also haveto account for 'trump' word mistakenly being used as a positive word. #####

### Finding the sum of all positive and negative words observed####

####  4. Performing sentiment analysis using the different word counts

In [107]:
if negative_counter > positive_counter:
    print("The general sentiment of tweets is negative")
elif positive_counter > negative_counter:
    print("The general sentiment of tweets is positive")
else:
    print("The general sentiment of tweets is neutral")

The general sentiment of tweets is negative


####Calculating ratio of positive, negative and stop word with respect to total words####

In [108]:
print("Ratio of positive to  whole word count = %.2f%%" %((positive_counter/total_count)*100))
print("Ratio of negative to  whole word count = %.2f%%" %((negative_counter/total_count)*100))
print("Ratio of stopwords to whole word count = %.2f%%" %((stop_counter/total_count)*100))

Ratio of positive to  whole word count = 2.32%
Ratio of negative to  whole word count = 3.86%
Ratio of stopwords to whole word count = 40.15%


In [109]:
#Calculating by what percentage is negative tweets exceeding positive tweets 
print ("Negative tweets increment by percentage value is %.2f%%" %(((negative_counter - positive_counter)/positive_counter) *100 ))

Negative tweets increment by percentage value is 66.22%


#### 5. Conclusion : Is the general sentiment positive or negative ? Strongly or weakly ? ###

#### The negative word counter exceeds the positive word counter. Ratio of negative to  whole word count is 3.86%. Negative tweets increment from positive tweets by percentage value is 66.22%. Therefore, the sentiment is strongly negative.


----------------

 ## HW1- Part 2 Analysing trending tweet - "Jeter" ###

#### 1. Data was collected from site https://trends24.in/united-states/  using Twitter API by running tweetering.py in spyder for  "Jeter"####

Importing re module for data manipulation

In [110]:
import re

In [111]:
data_clean=[] #creating an empty list
file=open("Jeter.txt","r") #open the Trump.txt file in read mode to analyze the twitter data.
##Looping through the file to convert the strings to lowercase,remove all leading and trailing white spaces.
for w in file: 
    w=w.lower()
    w=w.strip()
    w=w.replace("@","").replace(":","").replace('rt',"").replace(".","").\
    replace(",","").replace("!","").replace("?","").replace("#","").replace("\\","").\
    replace("\n","").replace("%","").replace("_","").replace("|","").replace("n\'","").\
    replace("-","").replace(')',"").replace("(","")
   
    for w in w.split():
       data_clean.append(w) #after splitting the strings append it to the data_clean list. 

In [112]:
#For further cleaning the data, using re.search function  
#creating a temporary empty list
a=[] 
#looping through the data which is cleaned above
for i in range(len(data_clean)): 
#re.search function will search for the mentioned characters
    if(re.search(r"https|xe2|xf0|[0-9]",data_clean[i]) is not None):    
#if the match is true(not None)]will append the location of item to list a
      a.append(i)

In [113]:
data_clean=[data_clean[i] for i in range(len(data_clean)) if (i not in a)] 

In [114]:
print (data_clean[:100])

['b"', 'sposcenter', 'seasons', 'in', 'baseball', 'hall', 'of', 'fame', 'inductee', 'derek', "jeter's", 'career', 'was', 'full', 'of', 'memorable', 'moments', 'mlb', 'derek', 'jeter', 'inspired', 'a', 'generation', 'of', 'shostops', 'basedlockha', 'derek', 'jeter', 'is', 'clearly', 'a', 'unanimous', 'hall', 'of', 'famer', 'look', 'at', 'his', 'stats', 'sposcenter', 'seasons', 'in', 'baseball', 'hall', 'of', 'fame', 'inductee', 'derek', "jeter's", 'career', 'was', 'full', 'of', 'memorable', 'moments', 'goodtweetman', 'no', 'one', 'ever', 'talks', 'about', 'how', 'george', 'is', 'just', 'ganking', 'dingersb"hey', 'everybody', 'derek', 'jeter', 'was', 'not', 'voted', 'in', 'unanimously', "it's", 'over', 'he', 'is', 'in', 'regardless', 'please', 'continue', 'with', 'your', 'dayn"b\'', 'without', 'george', 'constanza', 'who', 'knows', 'if', 'derek', 'jeter', 'makes', 'the', 'hall', 'of', 'famennvia', 'seinfeldtv']


#### 2. Using text files(with negative, positive and stop words) for comparison with cleaned data. 

In [115]:
positive =[]
f1= open('positive.txt','r')
for w in f1:
    w=w.lower()
    w=w.strip()
    positive.append(w)
len(positive)
positive
f1.close()

In [116]:
negative = []
f2=open('negative1.txt','r')
for x in f2:
    x=x.lower()
    x=x.strip()
    negative.append(x)
len(negative)
negative
f2.close()

In [117]:
stop=[]
f3 = open('stopwords1.txt','r')
for y in f3:
    y = y.lower()
    y = y.strip()
    stop.append(y)
len(stop)
stop[:2000]
f3.close()

#### 3. Comparing the lists above and performing analysis with the cleaned data

In [118]:
stop_counter = 0
negative_counter = 0
positive_counter = 0
others_counter = 0
total_count = 0

In [119]:
for w in data_clean:
    if w in positive:
        positive_counter +=1 
    elif w in negative:
            negative_counter +=1
    elif w in stop:
            stop_counter+=1
    else:
        others_counter+=1

In [120]:
total_count = positive_counter+negative_counter+stop_counter+others_counter
print("Total word count =%.2f" %(total_count))
print("positive word count =%d" %positive_counter)
print("negative word count =%d" %negative_counter)
print("stop word count =%d" %stop_counter)
print("other word count = %d" %others_counter)


print("Ratio of positive to  total word count = %.2f%%" %((positive_counter/total_count)*100))
print("Ratio of negative to  total word count = %.2f%%" %((negative_counter/total_count)*100))
print("Ratio of stopwords to total word count = %.2f%%" %((stop_counter/total_count)*100))

print("Ratio of other word v/s total words ratio = %.2f" %(others_counter/(total_count)))

print("Sum of all positive/negative words = %.2f" %(positive_counter+negative_counter))

print("Ratio of positive/(postive + negative words) = %.2f" %(positive_counter/(positive_counter+negative_counter)))

Total word count =25981.00
positive word count =969
negative word count =340
stop word count =9805
other word count = 14867
Ratio of positive to  total word count = 3.73%
Ratio of negative to  total word count = 1.31%
Ratio of stopwords to total word count = 37.74%
Ratio of other word v/s total words ratio = 0.57
Sum of all positive/negative words = 1309.00
Ratio of positive/(postive + negative words) = 0.74


#### 4. Analysing the word count

In [121]:
if negative_counter > positive_counter:
    print("The general sentiment of tweets is negative")
elif positive_counter > negative_counter:
    print("The general sentiment of tweets is positive")
else:
    print("The general sentiment of tweets is neutral")

The general sentiment of tweets is positive


In [122]:
print ("Positive tweets increment by percentage value is %.2f%%" %(((positive_counter - negative_counter)/negative_counter) *100 ))

Positive tweets increment by percentage value is 185.00%


#### 5. Conclusion : Is the general sentiment positive or negative ? Strongly or weakly ? ###

### The positive word counter exceeds the negative word counter. Ratio of positive to  total word count = 3.73%. Positive tweets increment negative tweets by percentage value of 185.00%. Therefore, the sentiment is extremely positive.#### 